bundles/org.openhab.binding.kaleidescape/src/main/java/org/openhab/binding/kaleidescape/internal/discovery/KaleidescapeDiscoveryService.java

```diff
@@ -281,7 +281,6 @@ private void isKaleidescapeDevice(String ipAddress) {
             input.close();
             writer.close();
             output.close();
-            socket.close();
```

Comment:
Why remove this?

Reply:
> Why remove this?

Socket inherits AutoClosable. Explicitly closing the socket not necessary when used in a try with resources block.